src/hooks/useInitialValues.ts

```diff
@@ -77,10 +77,39 @@ export function useInitialValues(
             formFields,
             encounterContext,
           );
+
+          if (field.questionOptions.calculate?.calculateExpression) {
+            evaluateAsyncExpression(
+              field.questionOptions.calculate.calculateExpression,
+              { value: field, type: 'field' },
+              formFields,
+              initialValues,
+              {
+                mode: encounterContext.sessionMode,
+                patient: encounterContext.patient,
+              },
+            )
+              .then((res) => {
+                // Assuming setInitialValues is a function that updates initialValues
+                if (
+                  res === existingVal ||
+                  (res && !existingVal) ||
+                  (res && field.questionOptions.defaultValue)
+                ) {
+                  setInitialValues((prevValues) => ({
+                    ...prevValues,
+                    [field.id]: res,
+                  }));
```

Comment:
I would use a similar approach used [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/hooks/useInitialValues.ts#L113-L123) for handling async operations or implement a better approach. If the calculated value is empty, we should fallback to picking the value from the encounter.

Reply:
Since the evaluation is asynchronous , the returned value will always be a promise